runtime/near-test-contracts/congestion-control-test-contract/src/lib.rs

```diff
@@ -76,6 +76,10 @@ extern "C" {
     // #######################
     fn promise_batch_action_create_account(promise_index: u64);
     fn promise_batch_action_deploy_contract(promise_index: u64, code_len: u64, code_ptr: u64);
+    fn promise_batch_action_deploy_global_contract(promise_index: u64, code_len: u64, code_ptr: u64);
```

Comment:
I assume deploys the global contract that is identifiable by its hash, right?

What should be the receiver_account_id for the promise? Anything? Or MUST it be the current_account_id (like the regular deploy action)?

Will it replace the currently deployed contract on the signer, predecessor, or receiver account id?

Reply:
> I assume deploys the global contract that is identifiable by its hash, right?

that is correct

> What should be the receiver_account_id for the promise? Anything? Or MUST it be the current_account_id (like the regular deploy action)?

receiver account id should be the same as current_account_id

> Will it replace the currently deployed contract on the signer, predecessor, or receiver account id?

`promise_batch_action_deploy_global_contract` doesn't replace anything, global contracts deployed by hash are not associated with any account. In case of `promise_batch_action_deploy_global_contract_by_account_id` it replaces current contract deployed under the signer account id.

Reply:
> I assume deploys the global contract that is identifiable by its hash, right?

that is correct

> What should be the receiver_account_id for the promise? Anything? Or MUST it be the current_account_id (like the regular deploy action)?

receiver should be the same as predecessor, see `check_actor_permissions` for `DeployGlobalContract` action

> Will it replace the currently deployed contract on the signer, predecessor, or receiver account id?

`promise_batch_action_deploy_global_contract` doesn't replace anything, global contracts deployed by hash are not associated with any account. In case of `promise_batch_action_deploy_global_contract_by_account_id` it replaces current contract deployed under receiver account id (which is required to be the same as predecessor).